File: dart_web_startup_sl8s/app/lib/model/example/example_wrapper/list_example_wrapper.dart
import 'package:dart_web_startup_sl8s/model/example/example.dart';
import 'package:dart_web_startup_sl8s/model/example/example_wrapper/example_wrapper.dart';
import 'package:dart_web_startup_sl8s/model/example/list_example.dart';
import 'package:library_architecture_mvvm_modify/library_architecture_mvvm_modify.dart';
import 'package:meta/meta.dart';

@immutable
base class ListExampleWrapper extends BaseListModelWrapper {
  const ListExampleWrapper({required List<List<dynamic>> listsListObject}) : super(listsListObject);

  @override
  ListExample createListModel() {
    final List<Example> listModel = List.empty(growable: true);
    for (final List<dynamic> itemListObject in listsListObject) {
      final exampleWrapper = ExampleWrapper(listObject: itemListObject);
      listModel.add(exampleWrapper.createModel());
    }
    return ListExample(listModel: listModel);
  }
}

File: typescript_discord_bot_startup_sl8s/app/src/model/example/example_wrapper/list_example_wrapper.ts
import { BaseListModelWrapper } from "@antonpichka/vitruvius";
import Example from "../example";
import ListExample from "../list_example";
import ExampleWrapper from "./example_wrapper";

class ListExampleWrapper extends BaseListModelWrapper {
    public constructor(listsListObject: Array<Array<any>>) {
        super(listsListObject);
    }
    
    public override createListModel(): ListExample<Example> {
        const listModel = new Array<Example>();
        for(const itemListObject of this.listsListObject) {
            const exampleWrapper = new ExampleWrapper(itemListObject);
            listModel.push(exampleWrapper.createModel());
        }
        return new ListExample(listModel);
    }
}

export default ListExampleWrapper;